File: mall-admin-web/src/api/tenant.ts
import { request } from '@/utils/request'

export interface Tenant {
  id: number | string
  tenantName: string
  contactName: string
  contactPhone: string
  contactEmail: string
  businessLicense: string
  licenseImage: string
  address: string
  status: number
  createTime: string
  scoreProduct: number
  scoreService: number
  scoreLogistics: number
  brandVerified: number
  commissionRate: number
  auditRemark: string
  logo: string
  description: string
}

export interface TenantApply {
  id: number | string
  tenantName: string
  contactName: string
  contactPhone: string
  contactEmail: string
  businessLicense: string
  licenseImage: string
  address: string
  applyStatus: number
  auditRemark: string
  auditTime: string
  auditUserId: number
  tenantId: number
  createTime: string
}

export interface TenantApplyParams {
  tenantName: string
  contactName: string
  contactPhone: string
  contactEmail: string
  businessLicense: string
  licenseImage: string
  address: string
}

// Tenant apply APIs (for audit)
export function getTenantApplyList(params?: { status?: number; page?: number; limit?: number }) {
  return request.get<{ list: TenantApply[]; total: number }>('/v1/platform/tenant/apply/list', params)
}

export function getTenantApplyDetail(id: number | string) {
  return request.get<TenantApply>(`/v1/platform/tenant/apply/${id}`)
}

export function auditTenantApply(id: number | string, pass: boolean, reason?: string) {
  return request.put(`/v1/platform/tenant/apply/${id}/audit`, null, { params: { pass, reason } })
}

export function getTenantList(params?: { status?: number; keyword?: string; page?: number; limit?: number }) {
  return request.get<{ list: Tenant[]; total: number }>('/v1/platform/tenant/list', params)
}

export function getTenantDetail(id: number | string) {
  return request.get<Tenant>(`/v1/platform/tenant/${id}`)
}

export function applyTenant(data: TenantApplyParams) {
  return request.post<number>('/v1/platform/tenant/apply', data)
}

export function auditTenant(id: number | string, pass: boolean, reason?: string) {
  return request.put(`/v1/platform/tenant/${id}/audit`, null, { params: { pass, reason } })
}

export function enableTenant(id: number | string) {
  return request.put(`/v1/platform/tenant/${id}/enable`)
}

export function disableTenant(id: number | string) {
  return request.put(`/v1/platform/tenant/${id}/disable`)
}

export function setTenantScore(id: number | string, data: { scoreProduct: number; scoreService: number; scoreLogistics: number; brandVerified: number }) {
  return request.put(`/v1/platform/tenant/${id}/score`, data)
}

// Merchant shop APIs (self-service, JWT-based)
export function getMerchantShop() {
  return request.get<Tenant>('/v1/merchant/shop')
}

export function updateMerchantShop(data: {
  tenantName: string
  logo?: string
  description?: string
  contactName?: string
  contactPhone?: string
  contactEmail?: string
  address?: string
}) {
  return request.put<void>('/v1/merchant/shop', data)
}

File: mall-model/src/main/java/com/mall/model/entity/Tenant.java
package com.mall.model.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.mall.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 商户表
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Tenant extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /** 商户名称 */
    private String tenantName;

    /** 联系人 */
    private String contactName;

    /** 联系电话 */
    private String contactPhone;

    /** 联系邮箱 */
    private String contactEmail;

    /** 营业执照号 */
    private String businessLicense;

    /** 营业执照图片 */
    private String licenseImage;

    /** 商户Logo */
    private String logo;

    /** 商户描述 */
    private String description;

    /** 状态: 0待审核 1已通过 2已拒绝 3已禁用 */
    private Integer status;

    /** 审核备注 */
    private String auditRemark;

    /** 店铺地址 */
    private String address;

    /** 佣金比例(百分比) */
    private java.math.BigDecimal commissionRate;

    /** 品牌认证: 0-否 1-是 */
    @TableField("brand_verified")
    private Integer brandVerified;

    /** 商品评分 */
    @TableField("score_product")
    private java.math.BigDecimal scoreProduct;

    /** 服务评分 */
    @TableField("score_service")
    private java.math.BigDecimal scoreService;

    /** 物流评分 */
    @TableField("score_logistics")
    private java.math.BigDecimal scoreLogistics;
}
